### PR TITLE
feat: add core workspace module

### PR DESCRIPTION
**Pull request by ruanyl**
### Description

The core workspace module(WorkspaceService) is a foundational component that enables the implementation of workspace features within OSD plugins. The purpose of the core workspace module is to provide a framework for workspace implementations.

This module does not implement specific workspace functionality but provides the essential infrastructure for plugins to extend and customize workspace features, it maintains a shared workspace state(observables) across the entire application to ensure a consistent and up-to-date view of workspace-related information to all parts of the application.

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by SuZhou-Joe — APPROVED** [review]

**Review by raintygao — APPROVED** [review]

File: src/core/public/core_system.ts [hunk 1]
```diff
@@ -110,6 +111,7 @@ export class CoreSystem {
 
   private readonly rootDomElement: HTMLElement;
   private readonly coreContext: CoreContext;
+  private readonly workspaces: WorkspaceService;
```
Comment (wanglam) [inline src/core/public/core_system.ts:114]: Do we need to rename variable to `workspace` or rename service name to `WorkspacesService`? Just find they are using consistent name in above.

File: src/core/public/core_system.ts [hunk 2]
```diff
@@ -242,6 +248,7 @@ export class CoreSystem {
         overlays,
         savedObjects,
         uiSettings,
+        workspaces,
```
Comment (ruanyl) [inline src/core/public/core_system.ts:251]: @wanglam I was trying to align the convention with the existing names.

Comment (ruanyl) [reply src/core/public/core_system.ts:251]: Oh, I get your point. You're right, I think I'd like change it to `workspace`, let me see how many places affected

**Review by wanglam — APPROVED** [review]

File: src/core/types/workspace.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ */
+
+export interface WorkspaceAttribute {
+  id: string;
+  name: string;
+  description?: string;
+  features?: string[];
+  color?: string;
+  icon?: string;
+  defaultVISTheme?: string;
```
Comment (Hailong-am) [inline src/core/types/workspace.ts:13]: include `reserved` also?

Comment (ruanyl) [reply src/core/types/workspace.ts:13]: Added

**Review by Hailong-am — APPROVED** [review]

**Pull request merged**